src/environment.ts

```diff
@@ -167,82 +158,96 @@ export class SecureEnvironment {
     remap(secureValue: SecureValue, rawValue: RawValue, rawDescriptors: PropertyDescriptorMap) {
         this.createSecureRecord(secureValue, rawValue);
         for (const key in rawDescriptors) {
-            // TODO: this whole block needs cleanup and simplification
-            // avoid overriding ecma script global keys.
-            if (!ESGlobalKeys.has(key)) {
-                const secureDescriptor = getOwnPropertyDescriptor(secureValue, key);
-                if (hasOwnProperty(rawDescriptors, key)) {
-                    // avoid poisoning to only installing own properties from baseDescriptors,
-                    let rawDescriptor = rawDescriptors[key];
-                    if (hasOwnProperty(rawDescriptor, 'set')) {
-                        // setter, and probably getter branch
-                        rawDescriptor = getSecureGlobalAccessorDescriptor(this, rawDescriptor);
-                    } else if (hasOwnProperty(rawDescriptor, 'get')) {
-                        // getter only branch (e.g.: window.navigator)
-                        const { get: originalGetter } = rawDescriptor;
-                        const env = this;
-                        rawDescriptor.get = function get(): SecureValue {
-                            const value: RawValue = apply(originalGetter as () => any, env.getRawValue(this), emptyArray);
-                            return env.getSecureValue(value);
-                        };
-                    } else {
-                        // value branch
+            // TODO: this whole loop needs cleanup and simplification avoid
+            // overriding ECMA script global keys.
+            if (ESGlobalKeys.has(key) || !hasOwnProperty(rawDescriptors, key)) {
+                continue;
+            }
+
+            // avoid poisoning to only installing own properties from baseDescriptors
+            const rawDescriptor = assign({}, rawDescriptors[key]);
+            setPrototypeOf(rawDescriptor, null);
+            if ('value' in rawDescriptor) {
```

Comment:
☝️ Here we shallow clone the descriptor since we're going to mutate it (also we `null` the prototype).